src/arithmetic/algebraic_expression.c

```diff
@@ -209,17 +221,48 @@ AlgebraicExpressionResult *AlgebraicExpression_Execute(AlgebraicExpression *ae)
 
         GrB_Descriptor_free(&desc);
     }
-        
+
+    if(ae->transpose) {
+        GrB_transpose(C, NULL, NULL, C, NULL);
+        // Swap src and dest nodes.
+        Node **n = ae->src_node;
+        ae->src_node = ae->dest_node;
+        ae->dest_node = n;
+    }
+
     AlgebraicExpressionResult *res = _AlgebraicExpressionResult_New(C, ae);
     return res;
 }
 
+void AlgebraicExpression_AppendTerm(AlgebraicExpression *ae, GrB_Matrix m, bool transpose) {
+    assert(ae);
+    ae->operand_count++;
+    ae->operands = realloc(ae->operands, sizeof(AlgebraicExpressionOperand) * ae->operand_count);
+    ae->operands[ae->operand_count-1].transpose = transpose;
+    ae->operands[ae->operand_count-1].operand = m;
+}
+
+void AlgebraicExpression_PrependTerm(AlgebraicExpression *ae, GrB_Matrix m, bool transpose) {
+    assert(ae);
+    ae->operand_count++;
+    ae->operands = realloc(ae->operands, sizeof(AlgebraicExpressionOperand) * ae->operand_count);
+    // TODO: might be optimized with memcpy.
```

Comment:
This would need `memmove` rather than `memcpy` - source and destination overlap.